frontend/src/metabase/querying/parameters/utils/query.ts

```diff
@@ -59,6 +59,10 @@ export function applyParameter(
     return query;
   }
 
+  if (stageIndex >= Lib.stageCount(query)) {
```

Comment:
Issue 2. We can get into a state where there is a parameter target with a stage index outside the query stages. E.g. in the referenced P1 issue we modify the query via the `underlying-records` drill, which drops query stages, and now post-aggregation parameters targets become invalid. We need to ignore them in this case.